test/functional/webvr.js

```diff
@@ -0,0 +1,275 @@
+
+var runtime = null;
+
+var rtLeft, rtRight;
+var _renderScale = 0.5; // 1
+
+var vrHMD = null;
+
+var frameData = null;
+if ('VRFrameData' in window) {
+  frameData = new VRFrameData();
+}
+
+var WebVRSupport = {};
+window.WebVRSupport = WebVRSupport;
+
+// defaults element IDs/defs, normally always passed
+var _viewpoint = "viewpoint";
+var _background = "background";
+var _scene = "scene";
+
+var _x3dEl = "x3d-elem";
+
+
+var viewpoint;
+var _initialPosition;
+
+/*
+options
+*/
+function _initialize( options ) {
+  _log('Initialize WebVR support');
+
+  if (options.viewpoint)
+    _viewpoint = options.viewpoint;
+  if (options.background)
+    _background = options.background;
+  if (options.scene)
+    _scene = options.scene;
+  if (options.renderScale)
+    _renderScale = options.renderScale;
+
+  if (options.x3dEl)
+    _x3dEl = options.x3dEl;
+
+  if (document.readyState === 'complete') {
+    load();
+  } else {
+    window.addEventListener('load', function ld(){
+      window.removeEventListener('load', ld);
+      load();
+    });
+  }
+}
+
+WebVRSupport.initialize = _initialize;
+
+function load() {
+  _log('Load external webvr.x3d dependency');
+
+  viewpoint = document.getElementById(_viewpoint);
+  if (viewpoint === null) {
+    console.log('viewpoint ID: ' + _viewpoint + ' not found');
+    return;
+  }
+  _initialPosition = viewpoint.getFieldValue('position');
+
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', 'webvr.x3d');
+  xhr.onreadystatechange = function() {
+    if (xhr.readyState === XMLHttpRequest.DONE) {
+      if (xhr.status === 200) {
+        var text = xhr.responseText;
+
+        text = text.replace(/\$VIEWPOINT/g, _viewpoint);
+        text = text.replace(/\$BACKGROUND/g, _background);
+        text = text.replace(/\$SCENE/g, _scene);
+
+        var node = document.createElement('group');
+        node.innerHTML = text;
+
+        var scene = document.getElementById('scene');
+        scene.appendChild(node);
+
+        init();
+      } else {
+        _log('error: could not load webvr.x3d');
+      }
+    }
+  }
+
+  xhr.send();
+}
+
+function init() {
+  runtime = document.getElementById(_x3dEl).runtime;
+
+  var ns = "";
+  ns = "Webvr__";
+  rtLeft = document.getElementById(ns+'rtLeft');
+  rtRight = document.getElementById(ns+'rtRight');
+
+
+  disableControls();
+
+
+  //runtime.enterFrame = enterFrame;
+
+  requestAnimationFrame(enterFrame);
+
+  // update viewpoint based on HMD pose
+  function enterFrame() {
+    if (!vrHMD) {
+      window.requestAnimationFrame(enterFrame);
+      return;
+    } else {
+      vrHMD.requestAnimationFrame(enterFrame); // native framerate if presenting
+    }
+
+    var state = getPose(vrHMD);
+
+    if (state.orientation !== null) {
+      var o = state.orientation;
+      var ori = viewpoint.requestFieldRef('orientation');
+      ori.x = o[0];
+      ori.y = o[1];
+      ori.z = o[2];
+      ori.w = o[3];
+      viewpoint.releaseFieldRef('orientation');
```

Comment:
viewpoint needs axis angle ? ok, orientation is probably a quaternion in the scenegraph.